Code/Framework/AzCore/AzCore/Script/ScriptAdapterComponent.cpp

```diff
@@ -0,0 +1,475 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2021-2021. All rights reserved.
+ * Description: ScriptAdapterComponent.
+ * Create: 2021-06-28
+ */
+#include <AzCore/Script/ScriptAdapterComponent.h>
+#include <AzCore/Script/ScriptSystemBus.h>
+#include <AzCore/Script/lua/lua.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/RTTI/RTTI.h>
+#include <AzCore/RTTI/BehaviorContextUtilities.h>
+#include <AzCore/Component/ComponentApplication.h>
+#include <AzCore/StringFunc/StringFunc.h>
+#include <AzCore/Utils/Utils.h>
+#include <AzCore/IO/SystemFile.h>
+#if defined(WIN32) || defined(WIN64)
+#include <atlsecurity.h>
+#endif
+
+namespace AZ
+{
+    const char* ScriptAdapterComponent::RELATIVE_LUA_PATH = "gen\\LuaAPI\\";
+    const char* ScriptAdapterComponent::GEN_DIR_NAME = "gen";
+    const char* ScriptAdapterComponent::GLOBAL_LUA_FILE_NAME = "GlobalsStaticLuaData.lua";
+    const char* ScriptAdapterComponent::CLASS_LUA_FILE_NAME = "ClassesStaticLuaData.lua";
+    const char* ScriptAdapterComponent::EBUS_LUA_FILE_NAME = "EbusesStaticLuaData.lua";
+    const int ScriptAdapterComponent::NAME_BUFFER_SIZE = 64;
+
+    void ScriptAdapterComponent::Reflect(ReflectContext* context)
+    {
+        SerializeContext* serializeContext = azrtti_cast<SerializeContext*>(context);
+        if (serializeContext)
+        {
+            serializeContext->Class<ScriptAdapterComponent, Component>()->Version(1);
+        }
+    }
+
+    void ScriptAdapterComponent::Activate()
+    {
+        AdapterBus::Handler::BusConnect();
+    }
+
+    void ScriptAdapterComponent::Deactivate()
+    {
+        AdapterBus::Handler::BusDisconnect();
+    }
+
+    AZStd::string ScriptAdapterComponent::GetProjectPath()
+    {
+        ComponentApplication* app = nullptr;
+        EBUS_EVENT_RESULT(app, ComponentApplicationBus, GetApplication);
+        const char* engineRoot = app->GetEngineRoot();
+        AZ_Assert(engineRoot != nullptr, "app GetEngineRoot failed");
+        const auto projectName = Utils::GetProjectName();
+        return AZStd::string::format("%s\\%s\\", engineRoot, projectName.c_str());
+    }
```

Comment:
I am assuming this logic was implemented in Lumberyard 1.X.

The AZ::Utils namespace has a [GetProjectPath](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Utils/Utils.h#L77-L78) function in it that doesn't assume the project is located with the engine root, nor does the code assume that the directory of the project is the same as the project name.